src/herder/HerderSCPDriver.cpp

```diff
@@ -792,6 +780,21 @@ HerderSCPDriver::valueExternalized(uint64_t slotIndex, Value const& value)
         abort();
     }
 
+    // externalize may trigger on older slots:
+    //  * when the current instance starts up
+    //  * when getting back in sync (a gap potentially opened)
+    // in both cases it's safe to not proceed with the rest of the function
+    // as we're already tracking a more recent state
+    // Yet, still deliver externalize events to LedgerManager so it
+    // can process catchup state, if needed
+    if (slotIndex <= mApp.getHerder().getCurrentLedgerSeq())
+    {
+        TxSetFramePtr externalizedSet = mPendingEnvelopes.getTxSet(b.txSetHash);
+        LedgerCloseData ledgerData(slotIndex, externalizedSet, b);
+        mLedgerManager.valueExternalized(ledgerData);
```

Comment:
@MonsieurNicolas I can update it. My original approach was to make herder behave in the same way as master, but maybe a better way would be to split events in Herder into "should happen for every ledger", and "should happen only for future ledgers". Also, do we care about metrics for old ledgers? (recording externalize events etc)

Regarding the second comment, I'm not sure what is the benefit of triggering the next ledger, when we still have some future ledgers buffered? (LedgerManager would technically still be "out of sync", right?)

Reply:
yes @marta-lokhova that's what I'd prefer to do: separate what we want to do for any ledger vs most recent.
I don't think tracking metrics for those out of order ledgers is that important so the list of what to do with older ledgers is fairly limited.

second issue (not for this PR) is that we don't even trigger when we close the most recent ledger, we have to close the next ledger (so we're delayed by one ledger).

Reply:
ok @MonsieurNicolas I made the changes. A few things here: 
* Saving SCP state: when an older slot is received, its envelopes are saved. It looks like we track the most recent transitive quorum, so I skipped saving anything quorum-related for older slots (as we already track newer state, so it seems like the right thing to do?).
* to be more conservative, we currently only allow older slots when herder is in sync (I think that makes sense, since catchup can't start anyway?)
* I'm looking into ways we could be more defensive about this: I notice that a lot of functions make assumptions about last known consensus index, and wondering if there is some invariance we could add now that we're letting older slots to be processed.

Reply:
@MonsieurNicolas re: second issue 
I think this happens in master currently as well. LedgerManager does not go "in sync" right after catchup applied all buffered ledgers - it waits for the next ledger. Since LM is not in sync, Herder skips triggering next ledger.